mne/viz/tests/test_ica.py

```diff
@@ -110,6 +110,21 @@ def test_plot_ica_components():
     with pytest.raises(RuntimeError, match='fit the ICA'):
         ica.plot_components(1, ch_type='mag')
 
+    # test provided axes
+    ica = ICA(n_components=None)
+    ica_picks = _get_picks(raw)  # 9 channels
+    ica.fit(raw, picks=ica_picks)
```

Comment:
Any way you can get away with not doing another `.fit`? All these ICA fits add up in testing time

Reply:
I tried to, but the other one above only fits 2 components which are not enough for all the cases below.
However, I did check that it fits in less than a second.

Reply:
> However, I did check that it fits in less than a second.

Okay, just keep in mind the ideal target is much less than 1 sec when possible. `pytest -m 'not ultraslowtest'` collects 3863 items for me so every bit counts. But it can be tough to optimize for ICA so we can live with it here :+1: 

At some point it would probably be good to do a minimal set of `ica.fit`s and `copy()` the objects around implicitly via a new `ica_raw_9comp` fixture or so for example. But that can be for another PR :)

Reply:
That's fair, so done. It's easy enough to do for short test functions ;)